doc/devel/coding_style_guideline.rst

```diff
@@ -0,0 +1,119 @@
+.. _coding_style_guideline:
+
+===========================
+DIPY Coding Style Guideline
+===========================
+
+The main principles behind dipy_ development are:
+* **Robustness**: the results of a piece of code must be verified
+systematically, and hence stability and robustness of the code must be ensured,
+reducing code redundancies.
+* **Readability**: the code is written and read by humans, and it is read much
+more frequently than it is written.
+* **Consistency**: following these guidelines will ease reading the code, and
+will make it less error-prone.
+* **Documentation**: document the code. Documentation is essential as it is one
+of the key points for the adoption of DIPY as the toolkit of choice in
+diffusion by the scientific community. Documenting helps clarifying certain
+choices, helps avoiding obscure places, and is a way to allow other members
+*decode* it with less effort.
+* **Language**: the code must be written in English. Norms and spelling should
+be abided by.
+
+------------
+Coding style
+------------
+
+DIPY follows the numpy_ coding style, described in their
+`HOWTO_DOCUMENT.rst.txt 
+`https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_
+file. Hence, DIPY uses the standard Python `PEP8
+<https://www.python.org/dev/peps/pep-0008/>`_ style to ensure the
+readability and consistency across the toolkit. Conformance to the PEP8 syntax
+is checked automatically when requesting to push to DIPY. There are
+`software systems <https://pypi.python.org/pypi/pep8>`_ that will check your code
+for PEP8 compliance, and most text editors can be configured to check the
+compliance of your code with PEP8. Beyond the aspects checked, as a contributor
+to DIPY, you should try to ensure that your code, including comments, conform to
+the above principles.
+
+-------------
+Documentation
+-------------
+DIPY uses `Sphinx <http://www.sphinx-doc.org/en/stable/index.html>`_ to generate
+documentation. We welcome contributions of examples, and suggestions for changes
+in the documentation, but please make sure that changes that are introduced
+render properly into the HTML format that is used for the DIPY website.
+
+When documenting functions, modules, class or methods we follow the
```

Comment:
I see -- it's already here! No need for that sentence in the previous section, IMO.